packages/core/test/execution/nonce-management/get-nonce-sync-messages.ts

```diff
@@ -145,394 +144,499 @@ describe("execution - getNonceSyncMessages", () => {
-            "Example#MyContract"
-          ] as DeploymentExecutionState
-        ).networkInteractions[0] as OnchainInteraction
-      ).transactions[0].hash;
+          return {};
+        });
 
-      await assertNoSyncMessageNeeded({
-        ignitionModule: exampleModule,
-        deploymentState,
-        transactionCountEntries: {
-          [exampleAccounts[1]]: {
-            pending: pendingCount,
-            latest: latestCount,
-            number: () => safestCount,
+        await assertGetNonceSyncThrows(
+          {
+            ignitionModule,
+            transactionCountEntries: {
+              [exampleAccounts[1]]: {
+                pending: pendingCount1,
+                latest: latestCount1,
+                number: () => {
+                  throw new Error("Should not be called");
+                },
+              },
+              // We define this just to validate that it's ignored, as its the default sender, but it won't be used
+              [exampleAccounts[2]]: {
+                pending: 2,
+                latest: 2,
+                number: () => {
+                  throw new Error("Should not be called");
+                },
+              },
+            },
+            latestBlockNumber: 3,
           },
-        },
-        // We are saying that the inflight transaction is still in the mempool
-        getTransaction: (txHash: string) => {
-          if (txHash !== inflightTxHash) {
-            throw new Error(
-              `Mock getTransaction was not expecting the getTransaction request for: ${txHash}`
-            );
-          }
-
-          return { _kind: "FAKE_TRANSACTION" };
-        },
+          `IGN403: You have sent transactions from ${exampleAccounts[1]}. Please wait until they get 5 confirmations before running Ignition again.`
+        );
       });
     });
+  });
 
-    it("should indicate the user replaced the transaction and the user transaction is safely confirmed", async () => {
-      // Set latest to be an arbitrary nonce
-      const latestCount = 30;
-      // Set safest to be the same as latest, it is not relevant
-      const safestCount = latestCount;
-      // There are no pending transactions
-      const pendingCount = latestCount;
-      // Set the nonce to be less than latest, indicating it was replaced
-      const nonce = latestCount - 1;
-
-      await assertGetNonceSyncResult(
-        {
-          ignitionModule: exampleModule,
-          deploymentState:
-            setupDeploymentStateBasedOnExampleModuleWithOneTranWith(nonce),
-          transactionCountEntries: {
-            [exampleAccounts[1]]: {
-              pending: pendingCount,
-              latest: latestCount,
-              number: () => safestCount,
+  describe("second deployment run", () => {
+    const latestBlockNumbers = [0, 3, 10];
+
+    for (const latestBlockNumber of latestBlockNumbers) {
+      describe(`With latest block ${latestBlockNumber}`, () => {
```

Comment:
```suggestion
      describe(`With latest block #${latestBlockNumber} and ${requiredConfirmations} required confirmations`, () => {
```

To make explicit that it is the no of confirmations interacting with the block number that makes each of these cases.